tests/queries/0_stateless/01917_distinct_on.sql

```diff
@@ -0,0 +1,9 @@
+DROP TABLE IF EXISTS t1;
+
+CREATE TABLE t1 (`a` UInt32, `b` UInt32, `c` UInt32 ) ENGINE = Memory;
+INSERT INTO t1 VALUES (1, 1, 1), (1, 1, 2), (2, 2, 2), (1, 2, 2);
+
+SELECT DISTINCT ON (a, b) a, b, c FROM t1;
+
```

Comment:
This query not ok
```
SELECT DISTINCT ON(a, b) * FROM AS t1
```

```
Received exception from server (version 21.7.1):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: 'b' 'FROM' 'a' 't1' while processing query: 'SELECT t1 LIMIT 1 BY (a, b) * FROM', required columns: 't1' 'a' 'FROM' 'b'.
```

Reply:
This query works, but looks wired, maybe should not allow such syntax without brackets?
```
:) SELECT DISTINCT ON a, b a, b FROM t1;

SELECT
    a,
    b
FROM t1
LIMIT 1 BY
    a,
    b

Query id: 5ebad59a-911a-4932-beae-20ef75a1a157

┌─a─┬─b─┐
│ 1 │ 1 │
│ 2 │ 2 │
│ 1 │ 2 │
└───┴───┘
```

Reply:
Also not sure should we forbid `DISTINCT ON` and `DISTINCT`
```
:) SELECT DISTINCT ON  (a, b) DISTINCT a, b FROM t1;

SELECT DISTINCT
    a,
    b
FROM t1
LIMIT 1 BY (a, b)

Query id: f57b59e3-3baf-4874-b00a-6e452b21d439

┌─a─┬─b─┐
│ 1 │ 1 │
│ 2 │ 2 │
│ 1 │ 2 │
└───┴───┘
```

Maybe it's ok

Reply:
Oh, I aimed to have 3 separate conversations for this cases, but it all pinned to one line of code 😕

Reply:
For the first case, I don't know how to deal with it.
For the second case, do you mean we should add a line `if (open_bracket.ignore(pos, expected))` before L114.
For the third case, it seems that we can allow both `DISTINCT ON` and `DISTINCT`?

Reply:
> For the first case, I don't know how to deal with it.

Ok, I'll try to figure it out and will suggest to you.

> For the second case, do you mean we should add a line if (open_bracket.ignore(pos, expected)) before L114.

Also need investigation.

> For the third case, it seems that we can allow both DISTINCT ON and DISTINCT?

I'm not sure. If we perform `DISTINCT ON` on any columns subset it implies `DISTINCT`, so it looks redundant. Maybe there's some edge case that I don't notice?

Reply:
> For the first case, I don't know how to deal with it.

Oh, sorry, I pasted parsed query. Original query was `SELECT DISTINCT ON (a, b) * FROM t1` but it wasn't parsed correctly. I fixed it and added to test.
 
> For the second case, do you mean we should add a line if (open_bracket.ignore(pos, expected)) before L114.

You are right, see my commit.

> For the third case, it seems that we can allow both DISTINCT ON and DISTINCT?

Finally, decided to make this mutually exclusive.